cmake-proxies/libcurl/CMakeLists.txt

```diff
@@ -0,0 +1,82 @@
+include(ExternalProject)
+
+def_vars()
+
+set( CURL_DIR "${_INTDIR}/libcurl" )
+set( CURL_TAG "curl-7_76_0")
+
+set(CURL_CMAKE_ARGS 
+   -DCMAKE_INSTALL_PREFIX:STRING=${CURL_DIR} 
+   -DHTTP_ONLY:BOOL=On 
+   -DBUILD_CURL_EXE:BOOL=Off
+   -DCMAKE_USE_LIBSSH2:BOOL=Off
+)
+
+if( CMAKE_SYSTEM_NAME MATCHES "Windows" )
+   set(CURL_CMAKE_ARGS ${CURL_CMAKE_ARGS} 
+      -DCMAKE_USE_SCHANNEL:BOOL=On
+   )
+elseif( CMAKE_SYSTEM_NAME MATCHES "Darwin" )
+   set(CURL_CMAKE_ARGS ${CURL_CMAKE_ARGS} 
+      -DCMAKE_USE_SECTRANSP:BOOL=On
+      -DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=10.9 
+      -DCMAKE_OSX_ARCHITECTURES:STRING=x86_64
+   )
+elseif( CMAKE_SYSTEM_NAME MATCHES "Linux|FreeBSD" )
+
+endif()
+
+ExternalProject_Add(curl
```

Comment:
This will not work on some Linux distributions as it downloads sources during build (build systems do not allow internet connection during build).

Apart from that, this suffers from the [same issue as the custom wxwidgets fork](https://github.com/audacity/audacity/issues/625): Including custom sources where in reality you should build against a system provided version of the library

Reply:
Oh, my. This story gets weirder the deeper one crawls.

* Some days ago, "Muse Group buys Audacity" -> :confused:
* a few days - nothing happens -> :cold_sweat:
* fefe blogs about this PR -> :scream:
* It's about "embedding" G**gle -> :angry: 
* You start digging and find friends already all over it -> :weary:
* They dig out stuff like ***THIS*** -> :rage: :exploding_head: :boom: :meat_on_bone: :fried_shrimp: :bone: 

I am very, very disappointed, Muse Group. And sad. Very sad, too.

------
In other recognition, thanks @dvzrv for pointing it out! It's pretty telling of the whole thing, I'd say....

Reply:
A NixOS package maintainer here: side effects like downloading sources during build are problematic indeed for many build systems. I'd kindly ask you to reconsider this, otherwise several distributions need to patch this downstream.